libs/vkd3d/utils.c

```diff
@@ -385,10 +385,27 @@ static void vkd3d_init_depth_stencil_formats(struct d3d12_device *device)
     }
 }
 
+static void vkd3d_init_non_depth_stencil_formats(struct d3d12_device *device)
+{
+    struct vkd3d_format *formats = (void *) device->formats;
```

Comment:
Again, weird cast to `void*`

Reply:
Would  `(struct vkd3d_format *)` be okay for you? Imo it's better to ignore const once then to not have it all. This way we can at least make sure it's only modified here.

Reply:
If we're throwing away const correctness anywhere, we shouldn't have it.

Reply:
if we write to an array then it shouldn't be const.

Reply:
ofc the big brain strategy would be to have a function return a `const struct vkd3d_format *` which both allocates and initializes that array, using non-const internally.

Reply:
In that case you still have cast away const to free the memory.

Reply:
That's a pretty common trope in C though. I don't have a problem with that.